### ISSUE TITLE
questions about Ms and Fs

### ISSUE THREAD

**Issue by cernykisss** [seoungwugoh/STM#21]
Hallo! Thanks for your codes! I appreciate you very much!
After i read your codes carefuully,i've noticed that there are two tensors called Es and Ms.

` Es = torch.zeros_like(Ms)
            Es[:,:,0] = Ms[:,:,0]`

Could you tell me their accurate meaning?

wish you a good day

**Comment by sjb961121** [comment]
> Hallo! Thanks for your codes! I appreciate you very much!
> After i read your codes carefuully,i've noticed that there are two tensors called Es and Ms.
> 
> ` Es = torch.zeros_like(Ms) Es[:,:,0] = Ms[:,:,0]`
> 
> Could you tell me their accurate meaning?
> 
> wish you a good day

I have the same question,Have you solved it? Could you help me?💋

**Comment by seoungwugoh** [comment]
Hi @cernykisss @sjb961121 
Originally, Ms contains the ground-truth mask and Es contains estimations. Ms was used for training but it no longer effective during inference. So, in the current demo code, you can consider Ms is just legacy doing nothing.

**Comment by cernykisss** [comment]
Thanks for your reply！
I've read your paper and knew that there are a Siamese Encoder consisting of two identical encoders in the model.   
I'm still wondering the meaning of Encoder_q and Encoder_m in your model.py.

**Comment by seoungwugoh** [comment]
In the paper (STM, http://openaccess.thecvf.com/content_ICCV_2019/papers/Oh_Video_Object_Segmentation_Using_Space-Time_Memory_Networks_ICCV_2019_paper.pdf), no Siamese encoder is used. There are two different encoder each for query and memory image, respectively. You may confused with my previous work (RGMP, http://openaccess.thecvf.com/content_cvpr_2018/papers/Oh_Fast_Video_Object_CVPR_2018_paper.pdf)

**Comment by cernykisss** [comment]
답장 해주셔서 매우 감사합니다！ 나는 신문을 잘못 읽었다.

**Comment by cernykisss** [comment]
Hello! Sorry for interrupting you again. I've read your paper about STM. But the I want to know more about multi object segment. Your paper says 'More details are included in the supplementary materials'. Where is the supplementary materials? Could you give me a link?
Waiting for your answer!

**Comment by seoungwugoh** [comment]
@cernykisss You can download it through the download link in the GitHub page.
https://openaccess.thecvf.com/content_ICCV_2019/supplemental/Oh_Video_Object_Segmentation_ICCV_2019_supplemental.zip

**Comment by cernykisss** [comment]
thank you！
